### ISSUE TITLE
[dart2js] RTI need doesn't take closure equality into account

### ISSUE THREAD

**Issue by johnniwinther** [dart-lang/sdk#47054]
The RTI need computation in dart2js doesn't take equality of closure into account. This means that when two instantiations of the same function with the same type arguments are compared at runtime, they might not be considered equal.

For instance, the program will not see that `create` needs its type arguments and the test there fails:
```
class Class<T> {}

Class<T> create<T>() => new Class<T>();

equals(a, b) {
  if (a != b) throw '$a != $b';
}

test<T>(f) {
  Class<T> Function() g = create;
  equals(f, g);
}

main() {
  Class<int> Function() f = create;
  test<int>(f);
}
```

This is the underlying issue why `language/constructor/tear_off_test` fails.

**Comment by sigmundch** [comment]
sorry, my bad, the CL addresses a different issue and adds a test for this issue, but it doesn't address yet the closure equality problem.

**Comment by rakudrama** [comment]
Was fixed by d49d5a96f74ec9aa2b775b82acc3b83a546d7d73
